### PR TITLE
Upgrade Storybook to v7

### PR DESCRIPTION
**Pull request by stephenjwatkins**
- Upgrades Storybook to v7
- Switches configuration files to TypeScript and updates their structure
- Convert MDX to MDX+CSF
- Add AriaProps to button prop types to support `onPress`, etc
- Fix Icon types build issue

### CODE REVIEW

File: easy-ui-icons/vite.config.ts [hunk 1]
```diff
@@ -87,15 +87,14 @@ async function getSvgFilenameFromJson(filename: string) {
...
-    }
     declare const ${svgName}: ({
       title,
       titleId,
       ...props
-    }: React.SVGProps<SVGSVGElement> & SVGRProps) => JSX.Element;
+    }: {
+      title?: string;
+      titleId?: string;
+    } & React.SVGProps<SVGSVGElement>) => JSX.Element;
```
Comment (stephenjwatkins) [inline easy-ui-icons/vite.config.ts:97]: TypeScript has trouble creating downstream definitions when `SVGRProps` is used due to it being an interface internal to this file, so having the structure inline remedies this issue

File: easy-ui-react/src/Button/Button.mdx
Comment (stephenjwatkins) [inline easy-ui-react/src/Button/Button.mdx:1]: Storybook essentially generated the MDX + CSF files here with its MDX migrator. we may want to create a separate task at some point to figure out a better Storybook documentation structure overall but this at least replicates our existing pattern in the new v7 format

File: easy-ui-react/src/utilities/storybook.tsx [hunk 2]
```diff
@@ -38,3 +40,27 @@ function getTokensControl(tokenAliases: string[]) {
...
+  <div style={buttonStoryStyles}>
+    <Story />
+  </div>
+);
+
+export const ButtonStoryOnDarkBackgroundDecorator: Decorator = (Story) => (
+  <div style={{ ...buttonStoryOnDarkBackgroundStyles, ...buttonStoryStyles }}>
+    <Story />
+  </div>
+);
```
Comment (OskiTheCoder) [inline easy-ui-react/src/utilities/storybook.tsx:66]: **suggestion:** doesn't necessarily have to be done now, can be a follow up, but as you pointed out on a previous PR, we could probably reuse these decorators for other inline components. what are your thoughts on renaming to `inline` and `Inline`?

File: easy-ui-react/src/IconButton/IconButton.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import AddIcon from "@easypost/easy-ui-icons/Add";
```
Comment (OskiTheCoder) [inline easy-ui-react/src/IconButton/IconButton.stories.tsx:1]: for some reason im unable to view this story via the preview link, do you know why that might be?

Screen Shot 2023-04-11 at 8 41 08 AM

Comment (stephenjwatkins) [reply easy-ui-react/src/IconButton/IconButton.stories.tsx:1]: nice catch. had a typo in the story file. even with the experimental MDX language server enabled in VSCode, i'm still finding it lacking with catching errors like that :(

**Pull request merged**